test/scalar.jl

```diff
@@ -82,5 +82,16 @@ end
     @test num(v,u,p,t) ≈ val * u
 
     @test convert(Number, num) ≈ val
+
+    # Test scalar operator which expects keyword argument to update, modeled in the style of a DiffEq W-operator.
+    γ = ScalarOperator(0.0; update_func=(args...; dtgamma) -> dtgamma, accepted_kwarg_fields=(:dtgamma,))
+
+    dtgamma = rand()
+    @test γ(u,p,t; dtgamma) ≈ dtgamma * u
```

Comment:
@gaurav-arya ^

Reply:
It would be nice to be able to provide a default value as well.
```julia
D = DiagonalOperator(zeros(N); update_func=(d, u, p, t; diag = zeros(N)) -> (A .= p * t * diag), accepted_kwargs = (:diag,))

D(u, p, t; diag = diag)
D(u, p, t)
```

Reply:
@gaurav-arya thanks for clarifying that `f(; kw)` is equivalent to `f(; kw = kw)`. In that case, discard the above comment.

Reply:
We can indeed provide a default value to the kwarg. I could add this to the examples, although I'm not sure if I should because I don't know a well-motivated case where we would want to do this (we would need an operator that only needs partial information to updade its state)